dvc/remote/gdrive.py

```diff
@@ -324,10 +325,40 @@ def _create_remote_dir(self, parent_id, title):
 
     @gdrive_retry
     def _delete_remote_file(self, remote_id):
+        from pydrive2.files import ApiRequestError
+
         param = {"id": remote_id}
         # it does not create a file on the remote
         item = self.drive.CreateFile(param)
-        item.Delete()
+
+        try:
+            item.Trash() if self._trash_only else item.Delete()
+        except ApiRequestError as exc:
+            http_error_code = exc.error.get("code", 0)
```

Comment:
so, if code is always set, let's just use `error["code"]`?

Reply:
unfortunately, `code` key might be not available; only `exc.error` at the moment is initialized with empty dict `{}` in case if `code` is not available in original HttpError.

Reply:
kk, let's at least use None or 0 consistently here and above